tsdb/mapper.go

```diff
@@ -460,14 +509,16 @@ func (a tagSetCursors) Keys() []string {
 
 // newTagSetCursor returns a tagSetCursor
 func newTagSetCursor(m string, t map[string]string, c []*seriesCursor, d *FieldCodec) *tagSetCursor {
-	return &tagSetCursor{
+	tsc := &tagSetCursor{
 		measurement: m,
 		tags:        t,
 		cursors:     c,
 		decoder:     d,
-		keyBuffer:   make([]int64, len(c)),
-		valueBuffer: make([][]byte, len(c)),
 	}
+	q := make(rawPoints, 0)
+	heap.Init(&q)
+	tsc.pqueue = &q
```

Comment:
These 3 lines could be refactored into a `newRawPoints()` function that returns `*rawPoints`. They're duplicated in a couple places.